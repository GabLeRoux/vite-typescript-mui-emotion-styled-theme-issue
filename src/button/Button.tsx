// Button.tsx
import React from 'react';
import { StyledButton } from "./Button.styled";

interface ButtonProps {
    isDarkTheme: boolean;
    toggleTheme: () => void;
}

const Button: React.FC<ButtonProps> = ({ isDarkTheme, toggleTheme }) => (
    <StyledButton onClick={toggleTheme}>
        Switch to {isDarkTheme ? 'Light' : 'Dark'} Theme
    </StyledButton>
);

export default Button;
