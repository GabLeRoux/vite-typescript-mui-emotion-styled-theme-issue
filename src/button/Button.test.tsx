import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@emotion/react';

import { lightTheme } from '../themes/light.theme';
import Button from './Button';

describe('<Button />', () => {


    it('should render correctly with dark theme', () => {
        const toggleThemeMock = () => {};

        const x = lightTheme;

        const view = render(
            <ThemeProvider theme={lightTheme}>
                <Button isDarkTheme={true} toggleTheme={toggleThemeMock} />
            </ThemeProvider>
        ).baseElement;

        expect(view).toMatchSnapshot();
    });

    it('should render correctly with light theme', () => {
        const toggleThemeMock = () => {};

        const view = render(
            <ThemeProvider theme={lightTheme}>
                <Button isDarkTheme={false} toggleTheme={toggleThemeMock} />
            </ThemeProvider>
        ).baseElement;

        expect(view).toMatchSnapshot();
    });
});
