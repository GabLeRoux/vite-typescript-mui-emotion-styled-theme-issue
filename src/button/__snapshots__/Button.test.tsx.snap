// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Button /> > should render correctly with dark theme 1`] = `
<body>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium eja6qb0 css-1eznx67-MuiButtonBase-root-MuiButton-root-StyledButton"
      tabindex="0"
      type="button"
    >
      Switch to 
      Light
       Theme
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</body>
`;

exports[`<Button /> > should render correctly with light theme 1`] = `
<body>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium eja6qb0 css-1eznx67-MuiButtonBase-root-MuiButton-root-StyledButton"
      tabindex="0"
      type="button"
    >
      Switch to 
      Dark
       Theme
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</body>
`;
